include/avk/commands.hpp

```diff
@@ -1089,23 +1089,48 @@ namespace avk
 	{
 		avk::resource_argument<avk::semaphore_t> mWaitSemaphore;
 		avk::stage::pipeline_stage_flags mDstStage;
+		uint64_t mValue;
 	};
 
 	inline semaphore_wait_info operator>> (avk::resource_argument<avk::semaphore_t> a, avk::stage::pipeline_stage_flags b)
 	{
-		return semaphore_wait_info{ std::move(a), b };
+		return semaphore_wait_info{ std::move(a), b, 0 };
 	}
 
+	inline semaphore_wait_info&& operator| (uint64_t aValue, semaphore_wait_info&& aInfo)
+	{
+		aInfo.mValue = aValue;
+		return std::move(aInfo);
+	}
+
+	inline semaphore_wait_info& operator| (uint64_t aValue, semaphore_wait_info& aInfo)
+	{
+		aInfo.mValue = aValue;
+		return aInfo;
+	}

```

Comment:
Hmm, I'm not so sure if we should really abuse `operator|` for that (because generally, Auto-Vk's API already abuses operators quite a lot).

Should we maybe rather, instead of `aSrcSignalStage >> sem | aSignalValue` aim for something like `aSrcSignalStage >> sem.at_value(aSignalValue)`? 
What do you think?

Reply:
Fine by me. This seems like it would definitely be more understandable to read.

Reply:
Since the semaphore type to construct the `semaphore_wait_info` is `avk::resource_argument<avk::semaphore_t>` I was hesitant to add an explicit template specialization for `avk::semaphore_t` to `avk::resource_argument`, adding the `at_value` / `to_value` functions.

A return type for such a function would also need to be either a `std::pair`, or some additional `incomplete_semaphore_wait_info` struct type, which both seem weird to me.

My current solution is to just add the `at_value` / `to_value` functions to the wait / signal info structs. Constructing the info structs would thus look like this `(aSrcSignalStage >> sem).at_value(aSignalValue)`. If I'm missing something obvious or you can think of a better solution please let me know.